### PR TITLE
[19.05] Allow mapping over of expression tool params

### PR DESCRIPTION
**Pull request by mvdbeek**
This allows feeding a collection of parameters values to a parameter
value. This enables [galaxyproject/tools-iuc#2404](https://github.com/galaxyproject/tools-iuc/pull/2404#issuecomment-491740393)
for instance, where we normalize a collection based on a computed
parameter.

### CODE REVIEW

File: lib/galaxy/workflow/run.py [hunk 1]
```diff
@@ -369,15 +368,22 @@ def replacement_for_connection(self, connection, is_data=True):
...
+        data_inputs = (model.HistoryDatasetAssociation, model.HistoryDatasetCollectionAssociation, model.DatasetCollection)
+        if not is_data and isinstance(replacement, data_inputs):
+            if isinstance(replacement, model.HistoryDatasetAssociation):
+                if replacement.is_pending:
+                    raise modules.DelayedWorkflowEvaluation()
+                if not replacement.is_ok:
+                    raise modules.CancelWorkflowEvaluation()
             else:
-                raise modules.CancelWorkflowEvaluation()
+                pending = False
```
Comment (jmchilton) [inline lib/galaxy/workflow/run.py:381]: should this check if ``replacement.collection.populated`` and delay if not or is that happening somewhere else?

Comment (mvdbeek) [reply lib/galaxy/workflow/run.py:381]: good point!

**Review by jmchilton — APPROVED** [review]
I'm a bit scared this doesn't match semantics I've established in the CWL branch... but it seems to make sense in this context. If things are incompatible I'll just deal with that downstream with different mapping types or something. 👍 and thanks @mvdbeek.

**Comment by mvdbeek** [comment]
Do you have some commits I can look at, maybe there's another way to do this ?

**Comment by jmchilton** [comment]
I wouldn't worry about it for now, thanks though!

**Pull request merged**